src/compile/time.ts

```diff
@@ -1,92 +1,100 @@
 import {contains, range} from '../util';
 import {COLUMN, ROW, SHAPE, COLOR, Channel} from '../channel';
-import {TimeUnit} from '../timeunit';
+import {TimeUnit, containsTimeUnit} from '../timeunit';
 
 /** returns the smallest nice unit for scale.nice */
 export function smallestUnit(timeUnit): string {
   if (!timeUnit) {
     return undefined;
   }
 
-  if (timeUnit.indexOf('second') > -1) {
+  if (containsTimeUnit(timeUnit, TimeUnit.SECONDS)) {
     return 'second';
   }
 
-  if (timeUnit.indexOf('minute') > -1) {
+  if (containsTimeUnit(timeUnit, TimeUnit.MINUTES)) {
     return 'minute';
   }
 
-  if (timeUnit.indexOf('hour') > -1) {
+  if (containsTimeUnit(timeUnit, TimeUnit.HOURS)) {
     return 'hour';
   }
 
-  if (timeUnit.indexOf('day') > -1 || timeUnit.indexOf('date') > -1) {
+  if (containsTimeUnit(timeUnit, TimeUnit.DAY) ||
+      containsTimeUnit(timeUnit, TimeUnit.DATE)) {
     return 'day';
   }
 
-  if (timeUnit.indexOf('month') > -1) {
+  if (containsTimeUnit(timeUnit, TimeUnit.MONTH)) {
     return 'month';
   }
 
-  if (timeUnit.indexOf('year') > -1) {
+  if (containsTimeUnit(timeUnit, TimeUnit.YEAR)) {
     return 'year';
   }
   return undefined;
 }
 
 export function parseExpression(timeUnit: TimeUnit, fieldRef: string, onlyRef = false): string {
   let out = 'datetime(';
-  let timeString = timeUnit.toString();
 
   function get(fun: string, addComma = true) {
     if (onlyRef) {
       return fieldRef + (addComma ? ', ' : '');
     } else {
-      return fun + '(' + fieldRef + ')' + (addComma ? ', ' : '');
+      let res = '';
+      if (fun === 'quarter') {
+        res = 'floor(month(' + fieldRef + ')' + '/3)*3';
```

Comment:
why do you have to multiply by 3 at the end?   (If there is a good reason, add comment.)